metadata_service/proxy/gcp_data_catalog_proxy.py

```diff
@@ -0,0 +1,350 @@
+# Copyright Contributors to the Amundsen project.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import re
+from typing import Union, Dict, List, Any, Optional
+
+from amundsen_common.models.dashboard import DashboardSummary
+from amundsen_common.models.popular_table import PopularTable
+from amundsen_common.models.table import Table, Column, Source, ProgrammaticDescription
+from amundsen_common.models.user import User as UserEntity, User
+from google.api_core.client_options import ClientOptions
+from google.cloud import datacatalog_v1
+from google.cloud.datacatalog_v1.proto.datacatalog_pb2 import Entry
+
+from metadata_service.entity.dashboard_detail import DashboardDetail as DashboardDetailEntity
+from metadata_service.entity.description import Description
+from metadata_service.entity.resource_type import ResourceType
+from metadata_service.proxy import BaseProxy
+from metadata_service.util import UserResourceRel
+
+LOGGER = logging.getLogger(__name__)
+
+
+# @todo - consider moving base class with init to commons - it's exactly the same in metadatalibrary
+class GCPDataCatalogProxy(BaseProxy):
+    def __init__(self, *,
+                 project_id: str = '',
+                 credentials_file: Optional[str] = None,
+                 client: datacatalog_v1.DataCatalogClient = None,
+                 page_size: int = 10,
+                 **kwargs
+                 ) -> None:
+
+        self.scope = datacatalog_v1.types.SearchCatalogRequest.Scope()
+        self.scope.include_project_ids.append(project_id)
+
+        client_options = ClientOptions(scopes=[self.scope])
+
+        if credentials_file:
+            _client = datacatalog_v1.DataCatalogClient.from_service_account_file(credentials_file,
+                                                                                 client_options=client_options)
+        else:
+            _client = datacatalog_v1.DataCatalogClient(credentials=credentials_file, client_options=client_options)
```

Comment:
In case of credentials_file being Non I would write it explicitly datacatalog_v1.DataCatalogClient(credentials=None, client_options=client_options). Otherwise a code reader is wondering whether it is a bug